cmd/nerdctl/main.go

```diff
@@ -171,6 +171,7 @@ func initRootCmdFlags(rootCmd *cobra.Command, tomlPath string) (*pflag.FlagSet,
 	rootCmd.PersistentFlags().StringSlice("hosts-dir", cfg.HostsDir, "A directory that contains <HOST:PORT>/hosts.toml (containerd style) or <HOST:PORT>/{ca.cert, cert.pem, key.pem} (docker style)")
 	// Experimental enable experimental feature, see in https://github.com/containerd/nerdctl/blob/main/docs/experimental.md
 	AddPersistentBoolFlag(rootCmd, "experimental", nil, nil, cfg.Experimental, "NERDCTL_EXPERIMENTAL", "Control experimental: https://github.com/containerd/nerdctl/blob/main/docs/experimental.md")
+	rootCmd.PersistentFlags().String("host-gateway-ip", cfg.HostGatewayIP, "IP address that the special 'host-gateway' string in --add-host resolves to. Defaults to the IP address of the default bridge")
```

Comment:
Any reason not to just use the host IP?

Reply:
The default bridge IP isn't useful for rootless because the slirp is launched with `--disable-host-loopback` for the security reason
https://github.com/containerd/nerdctl/blob/cffdf87ff4d648a5344eea1406bb95ca3ad7eaa4/extras/rootless/containerd-rootless.sh#L123

Reply:
Thanks for the suggestion. I just tried to use host ip as default, as the [commit](https://github.com/containerd/nerdctl/pull/1978/commits/08487b034c19b906b13bda9912aa1b553a613300) I pushed.

I tested the built nerdctl in Lima. The returned host ip can't be used to access host (curl failed). 

The static [Lima guest IP (192.168.5.15)](https://github.com/lima-vm/lima/blob/25b137d4c54a5a251fb7cca4adf1892097e1a6f2/docs/network.md#guest-ip-192168515) can be used to access host (curl succeeded). However it seems like a loopback IP which was skipped by the loopback check.

Reply:
I tried in ubuntu linux host and it works. I think the host IP is VM has special setup. I will follow the route of making host IP the default as it should work for normal cases. VM users will need to use the config to set the their special IP.

Reply:
I finally used env variable the pass the host IP to nsenter. Let me know if you have different thoughts.